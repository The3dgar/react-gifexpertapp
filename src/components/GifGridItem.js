import React from "react";

export const GifGridItem = ({ id, title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} loading="lazy"></img>
      <p>{title}</p>
    </div>
  );
};
