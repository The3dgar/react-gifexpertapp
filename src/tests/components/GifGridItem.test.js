import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import {GifGridItem} from "../../components/GifGridItem"


describe("pruebas en GifGridItem", () => {
  const title = "A title"
  const url = "https://media3.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=562ec969n2fygr06xiovb4paoa0ztco08uwp4fhkezy89wha&rid=giphy.gif"
  const wrapper = shallow(<GifGridItem title={title} url={url}></GifGridItem>);

  // beforeEach(()=>{
  //   wrapper = shallow(<GifGridItem></GifGridItem>);
  // })

  test("should render GifGridItem", () => {

    expect(wrapper).toMatchSnapshot()
    console.log("Corriendo el test")
  });

  test("debe de tener un parrafo con el title", ()=> {
    const p = wrapper.find("p")
    expect(p.text()).toBe(title)
  })

  test("debe de tener un imagen con url igual a props", ()=> {
    const img = wrapper.find("img")
    
    expect(img.props().src).toBe(url)
    expect(img.props().alt).toBe(title)
  })

  test("debe de tener un div con clase animate__fadeIn", ()=> {
    const div = wrapper.find("div")

    expect(div.prop("className").includes("animate__fadeIn")).toBe(true)
  })


  
});
