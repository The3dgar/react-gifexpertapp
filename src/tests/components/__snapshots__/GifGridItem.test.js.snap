// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem should render GifGridItem 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    loading="lazy"
    src="https://media3.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=562ec969n2fygr06xiovb4paoa0ztco08uwp4fhkezy89wha&rid=giphy.gif"
  />
  <p>
    A title
  </p>
</div>
`;
